R/oc_bbox.R

```diff
@@ -0,0 +1,146 @@
+#' Bounding box list for opencage queries
+#'
+#' Create a bounding box list for opencage queries.
+#'
+#' @param xmin Minimum longitude (also known as \code{min lon},
+#'   \code{southwest_lng}, \code{west}, or \code{left}).
+#' @param ymin Minimum latitude (also known as \code{min lat},
+#'   \code{southwest_lat}, \code{south}, or \code{bottom}).
+#' @param xmax Maximum longitude (also known as \code{max lon},
+#'   \code{northeast_lng}, \code{east}, or \code{right}).
+#' @param ymax Maximum latitude (also known as \code{max lat},
+#'   \code{northeast_lat}, \code{north}, or \code{top}).
+#' @param data A \code{data.frame} containing at least 4 columns with \code{xmin},
+#'   \code{ymin}, \code{xmax}, and \code{ymax} values, respectively.
+#' @param bbox A \code{bbox} object, see \link[sf]{st_bbox}.
+#' @param ... Ignored.
+#'
+#' @return A list of bounding boxes, each of class \code{bbox}.
+#' @export
+#'
+#' @examples
+#' oc_bbox(-5.63160, 51.280430, 0.278970, 51.683979)
+#'
+#' xdf <-
+#' data.frame(
+#'   place = c("Hamburg", "Hamburg"),
+#'   northeast_lat = c(54.0276817, 42.7397729),
+#'   northeast_lng = c(10.3252805, -78.812825),
+#'   southwest_lat = c(53.3951118, 42.7091669),
+#'   southwest_lng = c(8.1053284, -78.860521)
+#' )
+#' oc_bbox(
+#'   xdf,
+#'   southwest_lng,
+#'   southwest_lat,
+#'   northeast_lng,
+#'   northeast_lat
+#' )
+#' \dontrun{
+#' # create bbox list column with dplyr
+#' library(dplyr)
+#' xdf %>%
+#'   mutate(bbox =
+#'     oc_bbox(
+#'       southwest_lng,
+#'       southwest_lat,
+#'       northeast_lng,
+#'       northeast_lat
+#'      )
+#'    )
+#' }
+#' \dontrun{
+#' # create bbox list from a simple features bbox
+#'   library(sf)
+#'   bbox <- st_bbox(c(xmin = 16.1, xmax = 16.6, ymax = 48.6, ymin = 47.9),
+#'     crs = 4326)
+#'   oc_bbox(bbox)
+#' }
+oc_bbox <- function(...) UseMethod("oc_bbox")
+
+#' @name oc_bbox
+#' @export
+oc_bbox.default <- function (xmin, ymin, xmax, ymax, ...){
+  bbox <- function(xmin, ymin, xmax, ymax) {
```

Comment:
why is this function defined inside another function?

Reply:
It is not used somewhere else. But I can also put it outside.